### ISSUE TITLE
Reload does not work with "user-local default: JAR manifest" command line shortener

### ISSUE THREAD

**Issue by spand** [ktorio/ktor#1266]
**Ktor Version and Engine Used (client or server and name)**
Ktor 1.2.3 embedded Jetty server.

**Describe the bug**
If server is started with "user-local default: JAR manifest" command line shortener it seems `allUrls` in `ApplicationEngineEnvironmentReloading.kt` only has a couple of java jars thus it prints:

```
2019-08-07 14:11:13,088 INFO  --- [main] Application : No ktor.deployment.watch patterns match classpath entries, automatic reload is not active
```

**To Reproduce**
Error is shown on startup with the mentioned shorten command line option.

**Expected behavior**
Reloading works. It works for us if we use the `classpath file` shorten command line option. The options are not visible for kotlin run configurations but can be accessed in a Java Application run application (with same options as a Kotlin version.). I think the broken option is the default so it would be much nicer if Ktor worked with that.

**Comment by spand** [comment]
My colleague took another stab at this while doing some related work and discovered that the `@argfile` support from jdk9+ will be used by default and it seems to work with the reloading code as is.

Updating to jdk9+ is an acceptable "workaround" for us.

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-575">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Finally fixed in `1.5.3`
